File: server/src/messages/win.ts
import Player from "@/players/player";

/**
 * A message that is sent when a player wins a game
 *
 * Sent to both players in the session.
 */
export interface WinMessage {
  type: "win";
  player: string;
}

/**
 * Inform the players that a someone has won a game.
 *
 * Sent to both players in the session.
 * @param {Player} player The player who won
 * @returns {WinMessage} A WIn message
 */
function Win(player: Player): WinMessage {
  return {
    type: "win",
    player: player.name,
  };
}

export default Win;

File: dotnet/src/Connection/Win.cs
namespace Pong
{
  partial class Connection
  {
    /// <summary>
    /// A message that is sent when a player wins in session
    /// <br/>
    /// Sent to both players in the session
    /// </summary>
    public class WinEvent : GameEvent
    {
      public string Player { get; set; }

      public WinEvent(string player)
        : base("win")
      {
        Player = player;
      }

      /// <summary>
      /// Deserialize a WinEvent from json
      /// </summary>
      public static new WinEvent Deserialize(string json)
      {
        return Deserialize<WinEvent>(json);
      }
    }
  }
}
